internal/xds/translator/route.go

```diff
@@ -222,41 +222,70 @@ func buildXdsStringMatcher(irMatch *ir.StringMatch) *matcherv3.StringMatcher {
 	return stringMatcher
 }
 
-func buildXdsRouteAction(backendWeights *ir.BackendWeights) *routev3.RouteAction {
+func buildXdsRouteAction(backendWeights *ir.BackendWeights, settings []*ir.DestinationSetting) *routev3.RouteAction {
 	// only use weighted cluster when there are invalid weights
```

Comment:
Nit: this comment should be updated. Adding a few lines to explain the reason would help people who come later.